### PR TITLE
feat: issue support

### LINKED ISSUE #1683: Feature: Create an automated message once an issue is created
### Type of feature

📦 Chore

### Current behavior

Someone creates an issue and the needs triage issue label and other relevant label, e.g. feature, is added.

### Suggested solution

Keep doing the current behaviour, but consider generating an automated message along the lines of "Thanks for the issue, our team is looking into it. Once the issue is ready..."

Here is an example automated response from the Forem code base where I used to work. https://github.com/forem/forem/issues/11458#issuecomment-729173747

> Thanks for the issue! We'll take your request into consideration and follow up if we decide to tackle this issue.
>
> To our amazing contributors: issues labeled type: bug are always up for grabs, but for feature requests, please wait until we add a ready for dev before starting to work on it.
>
> To claim an issue to work on, please leave a comment. If you've claimed the issue and need help, please ping @forem/oss and we will follow up within 3 business days.
>
> For full info on how to contribute, please check out our [contributors guide](https://github.com/forem/forem/blob/master/CONTRIBUTING.md?rgh-link-date=2020-11-17T20%3A11%3A21Z).

Here is the workflow used to generate the automated message, https://github.com/forem/forem/blob/main/.github/workflows/issue.yml

I only mention it as I've seen several issues pop up where they're created and then the issue creator asks immediately to work on it. To save some time for the team (potentially), this automated flow would potentially prevent someone from asking to take the issue immediately before we consider it a valid issue. Or even if they ask immediately to take the issue, we won't need to respond because the automated message says we're looking into it.

Thoughts?

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by diivi**
## Description
Fixes #1683 

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents
#1683 

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

### CODE REVIEW

**Comment by diivi** [comment]
How do I test this? I can't create issues on my fork.

File: .github/workflows/issue.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      - name: Automatic Comment
+        uses: peter-evans/create-or-update-comment@v2
+        with:
+          issue-number: ${{ github.event.issue.number }}
+          body: |
+            Thanks for the issue, our team will look into it as soon as possible! If you would like to work on this issue, please wait for us to decide if it's ready. We will remove the "needs triage" label when it's ready to be worked on, and you can claim it then.
+
+            To claim an issue to work on, please leave a comment that says ".take". If you have any questions, please reach out to us on [Discord](https://discord.gg/U2peSNf23P) or follow up on the issue itself.
+
+            For full info on how to contribute, please check out our [contributors guide](https://docs.opensauced.pizza/contributing/introduction-to-contributing/).
```
Comment (nickytonline) [inline .github/workflows/issue.yml:28]: nit: add a new line at the end

File: .github/workflows/issue.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      pull-requests: write  # for peter-evans/create-or-update-comment to create or update comment
+    name: Comment
+    runs-on: ubuntu-latest
+    steps:
+      - name: Automatic Comment
+        uses: peter-evans/create-or-update-comment@v2
+        with:
+          issue-number: ${{ github.event.issue.number }}
+          body: |
+            Thanks for the issue, our team will look into it as soon as possible! If you would like to work on this issue, please wait for us to decide if it's ready. We will remove the "needs triage" label when it's ready to be worked on, and you can claim it then.
```
Comment (nickytonline) [inline .github/workflows/issue.yml:24]: wording (blocking): We should mention that the issue is available once the `needs triage` label has been removed.

Related to this, it'd be nice to update the `.take` GitHub action to block assignment if certain labels are present on the issue, e.g. `needs triage`

**Comment by nickytonline** [comment]
> How do I test this? I can't create issues on my fork.

The only way I know of testing a GitHub action like this is typically create a new repo and try it out there.

The testing story for GitHub actions isn't that great.

File: .github/workflows/issue.yml (hunk 2 above)
Comment (diivi) [reply .github/workflows/issue.yml:24]: I guess I'll have to make the second change in https://github.com/bdougie/take-action/blob/main/action.yml, I can block assignment but I don't think I can send a message about why it was blocked if I do this (if: part):

```
 take-issue:
    name: Disable take issue
    runs-on: ubuntu-latest
    timeout-minutes: 10
    if: "!contains(github.event.issue.labels.*.name, 'needs triage')"
    steps:
      - name: take an issue
        uses: bdougie/take-action@main
        with:
          issueCurrentlyAssignedMessage: Thanks for being interested in this issue. It looks like this ticket is already assigned to a contributor.
          token: ${{ secrets.GITHUB_TOKEN }}
```

**Review by nickytonline — APPROVED** [review]
🚢

**Review by brandonroberts — APPROVED** [review]
🚪

**Pull request merged**